textdb/textdb-api/src/main/java/edu/uci/ics/textdb/api/common/Schema.java

```diff
@@ -1,15 +1,47 @@
 package edu.uci.ics.textdb.api.common;
 
+import java.util.Arrays;
 import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 
 public class Schema {
-    private List<Attribute> attributes;
+    @Override

```

Comment:
In comparing two Tuples, along with comparing the fields we are also comparing schema. This is an additional check. Reference: DataTuple.java